plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/BaseTrinoCatalogTest.java

```diff
@@ -385,6 +386,61 @@ public void testView()
         }
     }
 
+    @Test
+    public void testListTables()
+            throws Exception
+    {
+        TrinoCatalog catalog = createTrinoCatalog(false);
+        TrinoPrincipal principal = new TrinoPrincipal(PrincipalType.USER, SESSION.getUser());
+        String ns1 = "ns1";
+        String ns2 = "ns2";
+
+        try {
+            catalog.createNamespace(SESSION, ns1, defaultNamespaceProperties(ns1), principal);
+            catalog.createNamespace(SESSION, ns2, defaultNamespaceProperties(ns2), principal);
+            SchemaTableName table1 = new SchemaTableName(ns1, "t1");
+            SchemaTableName table2 = new SchemaTableName(ns2, "t2");
+            catalog.newCreateTableTransaction(
+                            SESSION,
+                            table1,
+                            new Schema(Types.NestedField.of(1, true, "col1", Types.LongType.get())),
+                            PartitionSpec.unpartitioned(),
+                            SortOrder.unsorted(),
+                            arbitraryTableLocation(catalog, SESSION, table1),
+                            ImmutableMap.of())
+                    .commitTransaction();
+
+            catalog.newCreateTableTransaction(
+                            SESSION,
+                            table2,
+                            new Schema(Types.NestedField.of(1, true, "col1", Types.LongType.get())),
+                            PartitionSpec.unpartitioned(),
+                            SortOrder.unsorted(),
+                            arbitraryTableLocation(catalog, SESSION, table2),
+                            ImmutableMap.of())
+                    .commitTransaction();
+
+            // No namespace provided, all tables across all namespaces should be returned
+            List<SchemaTableName> allTables = catalog.listTables(SESSION, Optional.empty());
+            assertThat(allTables).containsAll(ImmutableList.of(table1, table2));
+            // Namespace is provided and exists
+            List<SchemaTableName> resultWithTable1 = catalog.listTables(SESSION, Optional.of(ns1));
+            assertThat(resultWithTable1).isEqualTo(ImmutableList.of(table1));
+            // Namespace is provided and does not exist
+            List<SchemaTableName> nonExistingNamespaceTables = catalog.listTables(SESSION, Optional.of("non_existing"));
+            assertThat(nonExistingNamespaceTables).isEmpty();
+        }
+        finally {
+            try {
+                catalog.dropNamespace(SESSION, ns1);
+                catalog.dropNamespace(SESSION, ns2);
+            }
+            catch (Exception e) {
```

Comment:
I see that this a pre-existing code template to swallow the exception and just provide a generic message (with no stacktrace), but why do we do this?

Why not let the exception bubble up so that we can eventually make some sense of it in the CI?

cc @alexjo2144 @findepi  - i see that this template has been introduced in 582b2d87

Reply:
I wonder what others think but to me it makes sense. This is a cleanup in the test itself.  If it fails, it may mislead folks when looking at CI that the test logic itself is failing due to the exception but really it's the cleanup. 

If the cleanup is required for state between tests, and it fails, then the tests would fail anyways.